### PR TITLE
Jomenxiao/add blackbox

### PR DESCRIPTION
**Pull request by jomenxiao**
add blackbox

### CODE REVIEW

File: roles/blackbox_exporter/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+
+# default configuration for node_exporter
```
Comment (LinuxGit) [inline roles/blackbox_exporter/defaults/main.yml]: blackbox_exporter

File: roles/blackbox_exporter/tasks/docker_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    mode: "0755"
+    backup: yes
+  with_items:
+    - run
+
+- name: load docker image from archive
+  docker_image:
+    state: present
+    force: yes
+    name: prom/node-exporter
```
Comment (LinuxGit) [inline roles/blackbox_exporter/tasks/docker_deployment.yml]: node-exporter ?

**Comment by LinuxGit** [comment]
Hi, you need to update rolling_update_monitor.yml, stop.yml, unsafe_cleanup.yml, unsafe_cleanup_container.yml.

File: scripts/blackbox_exporter.json [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+              "hide": false,
+              "intervalFactor": 2,
+              "refId": "A",
+              "step": 20
+            }
+          ],
+          "thresholds": [],
+          "timeFrom": null,
+          "timeShift": null,
+          "title": "Successful Hits Of  HTTP request Pushgateway ( test )",
```
Comment (LinuxGit) [inline scripts/blackbox_exporter.json]: need to remove ( test )

File: roles/blackbox_exporter/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+
+- include: "{{ deployment_method }}_deployment.yml"
+
+- name: prepare firewalld white list
```
Comment (LinuxGit) [inline roles/blackbox_exporter/tasks/main.yml]: put firewalld task at the end of the main.yml,   and be consistent with other roles.

File: scripts/blackbox_exporter.json (hunk 3 above)
Comment (jomenxiao) [reply scripts/blackbox_exporter.json]: I don't think so.
It means test, not real HTTP request.

**Comment by LinuxGit** [comment]
need to add blackbox_exporter in `name: import grafana dashboards` task of rolling_update_monitor.yml. @jomenxiao

File: roles/blackbox_exporter/tasks/docker_deployment.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    backup: yes
+  with_items:
+    - run
+
+- name: load docker image from archive
+  docker_image:
+    state: present
+    force: yes
+    name: prom/blackbox_exporter
+    tag: "{{ node_exporter_tag }}"
```
Comment (LinuxGit) [inline roles/blackbox_exporter/tasks/docker_deployment.yml]: blackbox_exporter_tag

**Pull request merged**